kwant_to_openfermion.py

```diff
@@ -17,25 +67,37 @@ def system_to_FermionOperator(sys):
         The hamiltonian of sys as an openfermion object.
     '''
     
-    if not isinstance(sys, kwant.system.System):
-        raise TypeError(f'Expecting an instance of System, got {type(sys)}')
+    
+    #Get the number of spin states out of the first on-site value
+    sample_val = sys.onsites[0][0]
+    if isinstance(sample_val, (int, float, complex)):
+        n_spin = 1
+    elif isinstance(sample_val, (numpy.ndarray, tinyarray.ndarray_complex, tinyarray.ndarray_float, tinyarray.ndarray_int)):
+        n_spin = sample_val.shape[0]
+    else:
+        raise TypeError(f'Expected a number or a numeric array, got {type(sample_val)}')
+    
+    
     
     ham = openfermion.FermionOperator()
 
     #on site terms
-    for ix in sys.id_by_site.values():
-        val = sys.hamiltonian(ix, ix)
-        if not isinstance(val, (int, float, complex)):
-            raise ValueError('Currently only supporting numeric on-site terms.')
-        ham = ham + openfermion.FermionOperator(f'{ix}^ {ix}', val)
+    for lat_ix in sys.id_by_site.values():
+        
+        val = sys.hamiltonian(lat_ix, lat_ix)
+        _check_dimension(val, n_spin)
+        
+        ham+=_single_term_to_FermionOperator(val, lat_ix, lat_ix, n_spin)
```

Comment:
Nit: spacing around the += (maybe try running the file through https://www.pylint.org/ )